### ISSUE TITLE
Decouple justpy from the addons

### ISSUE THREAD

**Issue by AgarwalPragy** [elimintz/justpy#17]
This framework looks promising, but it seems to be highly opinionated.
We can turn off the addons by setting the justpy.env file, but it seems that even with HIGHCHARTS, AGGRID, TAILWIND and QUASAR disabled, the JS and CSS for them are still loaded in the frontend.

Furthermore, fontawesome, animae, hover and simplemde CSS is loaded after the page's css, which makes overriding styles a pain.

It would be nice to be able to configure justpy in a manner that leaves all these addons out.

**Comment by elimintz** [comment]
As far as I can see, disabling HIGHCHARTS, AGGRID, TAILWIND and QUASAR does not load their css and JS. Please check again and send me an example, could be a bug.

I will add the option of leaving all addons out.

**Comment by fxnut** [comment]
Hi Eli,
Firstly, let me just say that speaking as someone who's principally a Python dev, I adore this framework. It makes it super easy to knock up interactive websites in Python. So thanks so much for all your time spent. I don't mind that you've made the decisions that you have, as it means that I don't have to spend all that time trying to form an opinion myself!

I just wanted to expand on the original issue, but for different reasons. I work in a secure network environment where open access to the internet cannot always be assumed. So that means I can't easily use JustPy in a professional environment as it relies on external CDNs to pull the some of the components. So I'd agree with @AgarwalPragy in that being able to have some control over these components would be massively useful. In my case, I'd like to be able to source them locally. 

It would also be useful, for when I'm developing while commuting, and internet access is not guaranteed.

I'd be hoping that this would also allow for custom build configurations for things like TailWind, as, by default, it doesn't include things like the "active:" variant modifier style, so unless I'm missing something, it's not currently possible to do button press styles out of the box without resorting to other tricks. 

The same goes for the ag-Grid component. At the moment, we can only use the Community version, but it'd be nice to be able to swap it out for the fully paid up version if required.

Maybe the easiest way to do this would be to allow a JustPy configuration setting to allow the developer to use a custom template directory? Then we could copy the existing template and modify it accordingly. But maybe there's a better way of achieving the same result without needing to effectively branch off the JustPy template?

So to summarise:
1) Allow local sourcing of components for no-internet operation
2) Allow use of local builds of components like TailWind, ag-Grid
3) Custom template directory

Anyway, I just thought I'd put these out to you for feedback. I'm happy to create separate Issues for you if you like.

Thanks again,
Andy

**Comment by elimintz** [comment]
Hi Andy,

Thank you for the great feedback and for using the framework. 

You can currently specify the template directory in the justpy.env file using the `TEMPLATES_DIRECTORY` variable. For example:
  
    TEMPLATES_DIRECTORY = '/my_template_dir'

You would then need to copy the templates directory of the package into your directory. It can be found at https://github.com/elimintz/justpy/tree/master/justpy/templates

Then you would need to change the templates as you see fit. 

This is currently cumbersome and requires delving into the templates. If you have a specific need I would be happy to help.

Another option is to for me to change justpy so that it includes all the JavaScript code and the templates do not need to access it via CDN. I think that makes sense also and I will work on that.

**Comment by fxnut** [comment]
Hi Eli,
That's great, thanks for the info! I must have overlooked the template directory variable, but yes, that would be great to have those dependencies brought into JustPy itself.

I have another question regarding mouse events, but I'll put that in a new issue. Thank you!

**Comment by fxnut** [comment]
Ah, never mind... I've just seen that my question was already discussed in [this issue](https://github.com/elimintz/justpy/issues/5). I was just after the shift/alt/ctrl status.
